R/best_case_worst_case_transition_risk_profile_at_company_level.R

```diff
@@ -105,18 +110,45 @@ prepare_for_join_at_company_level <- function(data) {
     distinct()
 }
 
-prepare_avg_worst_case_join_table <- function(data) {
-  data |>
-    select(-c("avg_transition_risk_best_case")) |>
-    filter(!is.na(.data$avg_transition_risk_worst_case))
+polish_transition_risk_best_case_worst_case_at_company_level <- function(data) {
+  rename(data, avg_transition_risk_equal_weight = "transition_risk_score_avg")
 }
 
-prepare_avg_best_case_join_table <- function(data) {
-  data |>
-    select(-c("avg_transition_risk_worst_case")) |>
-    filter(!is.na(.data$avg_transition_risk_best_case))
+add_avg_col <- function(data,
+                        avg_col,
+                        col,
+                        group_by,
+                        risk_category) {
+  mutate(data,
+    {{ avg_col }} := ifelse(is.na(.data[[col]]),
+      NA_real_,
+      mean(.data[[col]], na.rm = TRUE)
+    ),
+    .by = c(
+      col_companies_id(),
+      group_by,
+      risk_category
+    )
+  )
 }
 
-polish_best_case_worst_case_transition_risk_at_company_level <- function(data) {
-  rename(data, avg_transition_risk_equal_weight = "transition_risk_score_avg")
+add_avg_case_col_if_risk_category_match <- function(data,
+                                                    avg_case_col,
+                                                    risk_category,
+                                                    min_max_risk_category,
+                                                    avg_col) {
+  mutate(
+    data,
+    {{ avg_case_col }} :=
+      ifelse(data[[risk_category]] == data[[min_max_risk_category]],
```

Comment:
In line 143 do you really mean `data` or `.data`?

Reply:
Thanks for noticing! It should be `.data`. However, `data` was working fine too (I am not sure why that happens). I have changed it to `.data`.